src/typedarray.rs

```diff
@@ -318,6 +321,53 @@ macro_rules! typed_array_element {
     );
 }
 
+macro_rules! data_view_element {
+    ($t: ident,
+     $element: ty,
+     $unwrap: ident,
+     $length: ident) => (
+        pub struct $t;
+
+        impl TypedArrayElement for $t {
+            type Element = $element;
+            unsafe fn unwrap_array(obj: *mut JSObject) -> *mut JSObject {
+                $unwrap(obj)
+            }
+
+            unsafe fn length_and_data(obj: *mut JSObject) -> (u32,) {
+                ($length(obj),)
+            }
+
+        }
+    );
+
+    ($t: ident,
+     $element: ty,
+     $unwrap: ident,
+     $length: ident,
+     $create_new: ident,
+     $get_data: ident) => (
+        data_view_element!($t, $element, $unwrap, $length);
+        impl TypedArrayElementCreator for $t {
+            unsafe fn create_new(cx: *mut JSContext,array_buffer: *mut JSObject,byte_offset: u32,byte_length: u32) -> *mut JSObject {
```

Comment:
@jdm 
This code  creates a create_new function for data view object by extending the TypedArrayElement , and due to the mismatch in definition and prototype we are having errors like (Link to travis CI errors):

1.  [for the length and data method](https://travis-ci.com/servo/rust-mozjs/jobs/179728105#L1152 ) 
2.  [for the create new method](https://travis-ci.com/servo/rust-mozjs/jobs/179728105#L1142)
etc.

Reply:
Ah, you're right. I don't know if we can reuse the typed array support code for data views after all :(